src/bn.ts

```diff
@@ -0,0 +1,423 @@
+import BN from 'bn.js';
+
+export class BNWrapper<T extends BN> extends BN {
+  /**
+   * @description returns the maximum of 2 BN instances.
+   */
+  max(other: BN): T {
+    return BN.max(this, other) as T;
+  }
+
+  /**
+   * @description returns the minimum of 2 BN instances.
+   */
+  min(other: BN): T {
+    return BN.min(this, other) as T;
+  }
+
+  /**
+   * @description  clone number
+   */
+  clone(): T {
+    return super.clone() as T;
```

Comment:
Seems like most of these use the exact same syntax as BN. Can we use a [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) so our wrapper implementation doesn't drift from BN, and so it's more obvious how our implementation differs from theirs?

I think the syntax would be something like:

```ts
export const BNWrapper<T extends BN> = new Proxy(BN, {
  min(other: BN): T { … },
  …
})
```

Reply:
I just wanted it to return the new type. Which is why they all call `super.x`.  However, I'm now realizing that this solution doesn't work because it still doesn't supply the subclasses methods with a type assertion.